### PR TITLE
Add a high-level overview to README.md

### PR DESCRIPTION
**Pull request by kfogel**

### CODE REVIEW

**Comment by kfogel** [comment]
@jim-mcgowan (and Sam Caplan, if you're here -- sorry, I don't know your GitHub nick if you have one):

The easiest way to review this is probably to just look directly at the proposed [new version of the PDC Build README.md](https://github.com/PhilanthropyDataCommons/service/blob/noissue-overview-documentation/README.md).  Everything from *"This is the data-handling service layer..."* to *"See also the technical architecture diagram."* is the material I just added.  It is my attempt at a high-level description of what the PDC Build is all about.  Suggestions/edits welcome, and of course there might be some supplemental material we could add that doesn't necessarily belong in the README but would still be useful for conference presentations and such -- I'm happy to help draft such material, if you'd like.  Let me know.

**Comment by slifty** [comment]
Side note: there's clearly something up with our code coverage script -- I'll try to fix that separately but we can just ignore it for this PR since it's a documentation change

File: README.md
Comment (slifty) [inline README.md:10]: I know the target audience of this is the existing PDC community -- wondering if we should define "changemakers" somewhere for those who might not be in that loop? (e.g. we could create a glossary of terms in `/docs/GLOSSARY.md` and link to it?)

File: README.md
Comment (slifty) [inline README.md:17]: *gulp* we actually call these applications in PDC :( (proposal would be a more clear name IMHO, but we should probably be consistent)

**Review by slifty — APPROVED** [review]
Looks great -- one potential change is that you reference `proposals` but in PDC we call those `applications`.  Marked as approved since I'm not asking for the change so much as raising the possibility of the change.

I also wonder if we might want to make a GLOSSARY.md to define various terms of art.

File: README.md
Comment (kfogel) [reply README.md:10]: @slifty I like the idea of a glossary -- I don't think this PR needs to block on it, but it's a good idea.  The immediate audience is PDC folks, you are right, and they'll know the terms I'm using.  (The choice of "proposals" instead of "applications" was also deliberate, so that there would be no confusion between an application form and the submitted response to that form.  I realize that within the PDC code base "application" is the official term.)

File: README.md
Comment (kfogel) [reply README.md:17]: @slifty We can call them anything we want within the PDC code and documentation, but this high-level explanation has to be clear to outsiders.  We could add a parenthetical saying that within the code they're actually called "applications", but that that means "proposal".  Or we could just switch the PDC internally to using "proposal" (except I presume that there's some reason we didn't use that word in the first place, and whatever that reason is it still holds...?).

Comment (kfogel) [reply README.md:17]: @slifty "Application" by itself is ambiguous because it could mean either the blank form (the opportunity) or the submitted response that fills in the fields in the blank form (the proposal).  Even in the nearby presence of the word "opportunities", I'm worried that people -- especially those skimming -- might misunderstand, so I used "proposals" here.

Comment (kfogel) [reply README.md:17]: (Btw, I'll wait until this conversation seems to have resolved itself before merging... I realize you're not blocking the merge, but I want to make sure I'm not doing something dumb before I merge.)

Comment (slifty) [reply README.md:17]: Re application vs proposal: I don't remember the reasoning; it might have been someone said "proposal is not a good word to use for political reasons" or it might have been just the choice that was made and nobody applicationed -- err, proposed -- an alternative!

This is fine either way.  I think the bolding of it in the README is what threw me; it makes it sounds like "proposal" is our term for a system level / schema level / API endpoint level concept (opportunity, for instance is; as is field), and so it struck me as potentially confusing.

Comment (slifty) [reply README.md:17]: Looked at the history and indeed: it was originally `Proposal` and I believe Jim requested we make a change on behalf of stakeholders: https://github.com/PhilanthropyDataCommons/service/issues/2#issuecomment-1159186802

Comment (slifty) [reply README.md:17]: UPDATE: I'm implementing the application endpoint now -- I'm going to implement it with `proposals` since I like that better (and we can change it to application in review if needed)

Comment (kfogel) [reply README.md:17]: So should we merge this now, @slifty?

Comment (kfogel) [reply README.md:17]: Rebased just now; I'll merge.

**Pull request merged**
